code/modules/research/nanites/nanite_chamber.dm

```diff
@@ -25,8 +27,16 @@
 
 /obj/machinery/nanite_chamber/RefreshParts()
 	scan_level = 0
+	nanite_coeff = 0
+	speed_coeff = 1
 	for(var/obj/item/stock_parts/scanning_module/P in component_parts)
 		scan_level += P.rating
+	for(var/obj/item/stock_parts/manipulator/manipulator in component_parts)
+		nanite_coeff += manipulator.rating
+	var/total_laser_rating = 0
+	for(var/obj/item/stock_parts/micro_laser/micro_laser in component_parts)
+		total_laser_rating += micro_laser.rating
+	speed_coeff = 1 / (total_laser_rating * 0.5)
```

Comment:
```suggestion
	speed_coeff = 3 / (total_laser_rating + 2)
```

This is quite a steep curve here:
Tier 1: `1 / (1*0.5) = 2`
Tier 2: `1 / (2*0.5) = 1`
Tier 3: `1 / (3*0.5) = 0.66`
Tier 4: `1 / (4*0.5) = 0.5`

Try `speed_coeff = 3 / (total_laser_rating + 2)`

Tier 1: `3 / (1 + 2) = 1`
Tier 2: `3 / (2 + 2) = .75`
Tier 3: `3 / (3 + 2) = .60`
Tier 4: `3 / (4 + 2) = .5`

This also preserves the 1x multiplier for not being upgraded, so machines aren't suddenly working at half speed from roundstart, unless that was something you intended.

Reply:
Know that there's two micro-lasers per chamber, so the starting `total_laser_rating` would be 2.

`1 / (2 * 0.5)` = `1 / 1` = `1`

Reply:
Yep, my bad on that. 
That still makes for a very steep formula, but having been familiarized with exactly what both of these formulas affect, it's pretty inconsequential overall.